presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1391,7 +1392,7 @@ public PhysicalOperation visitIndexJoin(IndexJoinNode node, LocalExecutionPlanCo
         public PhysicalOperation visitJoin(JoinNode node, LocalExecutionPlanContext context)
         {
             List<JoinNode.EquiJoinClause> clauses = node.getCriteria();
-            if (clauses.isEmpty() && !node.getFilter().isPresent()) {
+            if (clauses.isEmpty() && !node.getFilter().isPresent() && node.getType() == INNER) {
```

Comment:
I think the best thing to do here is to modify the nested loop join code to be aware of the type of join so that it can emit nulls correctly.

Otherwise, this will force LJ, RJ, and FJ to use the hash build join even when the criteria is just TRUE. The nested loop join will be much more efficient in those cases.